realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -172,6 +172,11 @@ static synchronized void release(BaseRealm realm) {
             return;
         }
 
+        realm.tidyPerformedTransactions();
+        if (!realm.asyncTransactions.isEmpty() || !realm.asyncTransactionCallbacks.isEmpty()) {
```

Comment:
I tend to agree, but this arose from a user having problems because it wasn't clear that closing the Realm would also kill any notifications.

Instead of creating new lists keeping track of this. Couldn't we just extend the RealmThreadPoolExecutor to keep track of this?